vxmessenger/transport.py

```diff
@@ -183,45 +191,50 @@ def handle_raw_inbound_message(self, message_id, request):
             return
 
         try:
-            page = Page.from_fp(request.content)
-            self.log.info("MessengerTransport inbound %r" % (page,))
+            pages, errors = Page.from_fp(request.content)
         except (UnsupportedMessage,), e:
             self.respond(message_id, http.OK, {
                 'warning': 'Accepted unsuppported message: %s' % (e,)
             })
             self.log.error(e)
             return
 
-        if self.config.get('retrieve_profile'):
-            helper_metadata = yield self.get_user_profile(page.from_addr)
-        else:
-            helper_metadata = {}
-
-        yield self.publish_message(
-            message_id=message_id,
-            from_addr=page.from_addr,
-            from_addr_type='facebook_messenger',
-            to_addr=page.to_addr,
-            in_reply_to=page.in_reply_to,
-            content=page.content,
-            provider='facebook',
-            transport_type=self.transport_type,
-            transport_metadata={
-                'messenger': {
-                    'mid': page.mid,
-                }
-            },
-            helper_metadata={
-                'messenger': helper_metadata
-            })
+        self.log.info("MessengerTransport inbound %r" % (pages,))
+        if errors:
+            for error in errors:
+                self.log.error(error)
```

Comment:
wow, that is very silly of me. It should be either:
if errors:
    log errors
or:
for error in errors:
    log error

Which would you prefer? I think I started with the second one, then changed badly to the first.